pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -281,3 +307,151 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
+		Mtls: &v1beta1.PeerAuthentication_MutualTLS{
+			Mode: v1beta1.PeerAuthentication_MutualTLS_PERMISSIVE,
+		},
+	}
+
+	if meshPolicy != nil && meshPolicy.Mtls != nil && meshPolicy.Mtls.Mode != v1beta1.PeerAuthentication_MutualTLS_UNSET {
+		// If mesh policy is define, update parentPolicy to mesh policy.
+		parentPolicy.Mtls = meshPolicy.Mtls
+	}
+
+	if namespacePolicy != nil && namespacePolicy.Mtls != nil && namespacePolicy.Mtls.Mode != v1beta1.PeerAuthentication_MutualTLS_UNSET {
+		// If namespace policy is define, update parent policy to namespace policy. This means namespace
+		// policy can completely overwrite mesh policy.
+		parentPolicy.Mtls = namespacePolicy.Mtls
+	}
+
+	if numWorkloadLevelPolicies == 0 {
+		// There is no workload-level policy, return the parent policy.
+		return &parentPolicy
+	}
+
+	// Placeholder for the final consolidated policy.
+	finalPolicy := v1beta1.PeerAuthentication{
+		Mtls: &v1beta1.PeerAuthentication_MutualTLS{
+			Mode: v1beta1.PeerAuthentication_MutualTLS_UNSET,
+		},
+	}
+
+	// Second round to handle workload level policies. For each policy:
+	// - If its workload-level mTLS is not defined or is UNSET, treat it as if it has the parent mTLS.
+	// - If the policy mTLS is "strictly stronger" than the consolidation, update the consolidation's
+	// mTLS to that.
+	// - If the policy has port-level mTLS and is "strictly stronger" than the one in consolidation
+	// (if exists), copy it to consolidation.
+	// - If the port-level mTLS is UNSET, mark the port and do another round of reconcile with workload level.
+	unsetPorts := make(map[uint32]bool)
+	for _, cfg := range configs {
+		spec := cfg.Spec.(*v1beta1.PeerAuthentication)
+		// Ignore namespace and mesh level policies. Also ignore policies defined in root namespace
+		// but have selector (only mesh-level policy can be defined in root namespace).
+		if cfg.Namespace == rootNamespace || spec.Selector == nil {
+			continue
+		}
+
+		if spec.Mtls != nil &&
+			spec.Mtls.Mode != v1beta1.PeerAuthentication_MutualTLS_UNSET &&
+			isStrictlyStronger(spec, &finalPolicy) {
+			// Current policy has explicit mTLS, with stronger mTLS mode than the consolidated policy: update to current.
+			finalPolicy.Mtls = spec.Mtls
+		} else if isStrictlyStronger(&parentPolicy, &finalPolicy) {
+			// Current policy inherit from parent, and parent has stronger mTLS mode than the consolidate policy: update to parent.
+			finalPolicy.Mtls = parentPolicy.Mtls
+		}
+
+		// Check port level settings.
+		for port, mtls := range spec.PortLevelMtls {
+			if finalPolicy.PortLevelMtls == nil {
+				finalPolicy.PortLevelMtls = make(map[uint32]*v1beta1.PeerAuthentication_MutualTLS)
+			}
+
+			if mtls.Mode == v1beta1.PeerAuthentication_MutualTLS_UNSET {
+				unsetPorts[port] = true
+			} else {
+				existing, exist := finalPolicy.PortLevelMtls[port]
+				if !exist || getMutualTLSMode(mtls) > getMutualTLSMode(existing) {
+					finalPolicy.PortLevelMtls[port] = mtls
+				}
+			}
+		}
```

Comment:
You have to break out of the main for loop once you get a workload specific policy. Thsi will be the oldest policy. We cannot pick the strongest as it has the potential to break workloads that dont even belong to the team that wrote this policy.